executor.py

```diff
@@ -41,13 +42,16 @@ def index(
     def search(
         self,
         docs: Optional['DocumentArray'] = None,
+        parameters: Optional[Dict] = None,
         **kwargs,
     ):
         """Perform a vector similarity search and retrieve the full Document match
 
         :param docs: the Documents to search with"""
-
-        docs.match(self._storage, **self._match_args)
+        match_args = deepcopy(self._match_args)
+        if parameters and 'match_args' in parameters:
+            match_args.update(parameters['match_args'])
```

Comment:
```suggestion
        if parameters:
            match_args.update(parameters)
```

Reply:
i would in general consider `match_args`  always a runtime arg. so maybe delete it from the initialization, more radical but more correct. 

I'm also fine with the current approach

Reply:
`parameters` is now updated [here](https://github.com/jina-ai/executor-simpleindexer/pull/12/commits/9df6e2d235cc76a43b70bba6efbe0197d3c1c0b0) to avoid the argument nesting.

I see the benefits of deleting `match_args` from `__init__` as it is a runtime arg, however having some arguments like `traversal_ldarray` and `traversal_rdarray` kept also saves users from having to pass them to the query parameters every time they make a search request. If you're fine with current approach, I am more inclined to keep `match_args` because of its convenience.